lib/rack/utils.rb

```diff
@@ -211,6 +211,24 @@ def select_best_encoding(available_encodings, accept_encoding)
       (encoding_candidates & available_encodings)[0]
     end
 
+    # A <cookie-name> can be any US-ASCII characters, except control characters, spaces, or tabs. It also must not contain a separator character like the following: ( ) < > @ , ; : \ " / [ ] ? = { }.
+    # Cookies names starting with __Secure- (dash is part of the prefix) must be set with the secure flag from a secure page (HTTPS).
+    # Cookies with names starting with __Host- must be set with the secure flag, must be from a secure page (HTTPS), must not have a domain specified (and therefore aren't sent to subdomains) and the path must be /
```

Comment:
It's odd to have this documentation, but not actually check that these things are true.

Reply:
We do a little more validation in the follow up PR.